upgrade/upgrade-provider.go

```diff
@@ -305,6 +305,11 @@ func UpgradeProvider(ctx Context, repoOrg, repoName string) error {
 				"go", "get", "github.com/pulumi/pulumi/pkg/v3")).
 				In(repo.examplesDir()))
 	}
+	if ctx.UpgradePfVersion {
+		steps = append(steps, step.Cmd(exec.CommandContext(ctx,
+			"go", "get", "github.com/pulumi/pulumi-terraform-bridge/pf")).
+			In(repo.providerDir()))
+	}
```

Comment:
This should be similar to upgrading the bridge version (with an explicit target version) and probably occur right afterwards.